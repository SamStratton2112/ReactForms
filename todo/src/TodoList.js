import { useState } from "react";
import Todo from './Todo.js'
import NewTodoForm from './NewTodoForm.js';

const TodoList = () =>{
  const [todos, setTodos] = useState([]);
  const addTodo = (newTodo)=>{
    // spread all todos and add new one 
    setTodos(todos=>[...todos, newTodo])
  }
  const removeTodo = id=>{
    setTodos(todos=> todos.filter(t=>t.id!==id))
  }

  return(
    <div>
      <NewTodoForm addTodo={addTodo}/>
      <ul>
        {todos.map(todo=>{
          return(
            <li key={todo.id}>
              <Todo  id={todo.id} todo={todo.todo} removeTodo={removeTodo}/>
            </li>
          )
        })}
      </ul>
    </div>
  )
}

export default TodoList;