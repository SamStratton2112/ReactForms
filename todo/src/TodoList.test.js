import { render, fireEvent } from '@testing-library/react';
import TodoList from './TodoList';

it('renders', () => {
  render(<TodoList />);
});

it('should match snapshot', () => {
  const {asFragment}=render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it('should add a todo and then remove a todo', ()=>{
    const {queryByText} = render(<TodoList/>)
    const btn = queryByText('Add To Do')
    expect(queryByText('X')).not.toBeInTheDocument();
  
    fireEvent.click(btn)
    expect(queryByText('X')).toBeInTheDocument();
    // passes add todo portion of test 
  
    const rmvBtn = queryByText('X');
    fireEvent.click(rmvBtn);
    expect(queryByText('X')).not.toBeInTheDocument();
    // passes remove todo portion of test 
  })