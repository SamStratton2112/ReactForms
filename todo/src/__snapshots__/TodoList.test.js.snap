// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="todo"
      >
        To Do
      </label>
      <input
        id="todo"
        name="todo"
        placeholder="What to "
        type="text"
        value=""
      />
      <button>
        Add To Do
      </button>
    </form>
    <ul />
  </div>
</DocumentFragment>
`;
